crates/hir-def/src/hir/format_args.rs

```diff
@@ -177,9 +177,9 @@ pub(crate) fn parse(
     let text = s.text_without_quotes();
     let str_style = match s.quote_offsets() {
         Some(offsets) => {
-            let raw = u32::from(offsets.quotes.0.len()) - 1;
+            let raw = usize::from(offsets.quotes.0.len()) - 1;
             // subtract 1 for the `r` prefix
-            (raw != 0).then(|| raw as usize - 1)
```

Comment:
This is no longer lazy, so this always underflows when raw = 0

Reply:
Seems weird for clippy to lint this as arithmetic is side effectful in debug builds

Reply:
Yes, I am also a bit disappointed. But I guess in 99% of the cases doing it not lazy will have the correct behavior.